mne/epochs.py

```diff
@@ -3446,16 +3309,21 @@ def __init__(self, fname, proj=True, preload=True,
         # again, ensure we're retaining the baseline period originally loaded
         # from disk without trying to re-apply baseline correction
         super(EpochsFIF, self).__init__(
-            info, data, events, event_id, tmin, tmax, baseline=None, raw=raw,
+            info, data, events, event_id, tmin, tmax,
+            baseline=None, raw=raw,
             proj=proj, preload_at_end=False, on_missing='ignore',
             selection=selection, drop_log=drop_log, filename=fname_rep,
-            metadata=metadata, verbose=verbose, **reject_params)
+            metadata=metadata, verbose=verbose, raw_sfreq=raw_sfreq,
+            **reject_params)
         self.baseline = baseline
         self._do_baseline = False
         # use the private property instead of drop_bad so that epochs
         # are not all read from disk for preload=False
         self._bad_dropped = True
 
+        # set annotations
+        self.set_annotations(annotations)
```

Comment:
Maybe we should have `BaseEpochs.__init__(..., annotations=None)`. That would work well with my comment above that setting `self._annotations = None` should be handled by the base class.

Reply:
Agreed. Moved setters into BaseEpochs.